### ISSUE TITLE
Use `transparent` as a ColorVariable?

### ISSUE THREAD

**Issue by chris-dura** [causes/scss-lint#376]
```scss
.widget {
   border: 4px solid transparent;
}
```
throws this error:
```bash
[W] ColorVariable: Color literals like `transparent` should only be used in variable declarations; they should be referred to via variable everywhere else.
```
I guess I understand the "why"... I would certainly want to avoid using other `ColorKeywords` like `red` or `green`.

To get around this, I guess I have to write:
```scss

$transparent: transparent;
...

.widget {
   border: 4px solid $transparent;
}
```

Writing `$transparent: transparent;` seems a little overwrought to me, so just wondering if this is the expected solution, or should `transparent` be an exception to the Linter?

**Comment by bartveneman** [comment]
And what about currentColor, the same, I guess?

**Comment by JohnAlbin** [comment]
> or should transparent be an exception to the Linter?

First off, I find the new ColorVariable linter incrediably useful. It's helping me clean up a legacy code base with colors all over the place.

But when I use `transparent` in my stylesheet it is because I'm trying to _remove_ a color that is cascading or from another ruleset. Creating a variable called `$this-element-border: transparent;` obfuscates the fact that the element does NOT have a border. Similarly, creating a $no-color or $transparent variable is "overwrought" as @chris-dura said.

Unfortunately, if ColorVariable continues to complain about `transparent` needing to be in a variable, I'm going to have disable this linter after I finish my refactoring for today. And raw colors will inevitably sneak back into the stylesheets. :-p

**Comment by sds** [comment]
Which versions of `scss-lint` and `sass` are you using, @chris-dura? 

I don't see the above behavior with the `ColorVariable` linter with the latest `scss-lint` (0.34.0) and sass (3.4.13).

**Comment by chris-dura** [comment]
Sass 3.4.3 (Selective Steve)
scss-lint 0.34.0

Sorry, my example was pseudo-code, not straight from my codebase... and my example code, as you said, is not throwing an error. So, it must be something else *around* that line in my actual code causing that line to report a lint error... 

I'll troubleshoot my codebase and see if the error is actually indicative of some other bug. Feel free to close this issue or leave it open.

**Comment by JohnAlbin** [comment]
```
$ bundle exec scss-lint sass/
sass/components/_progress-bar.scss:52 [W] ColorVariable: Color literals like `transparent` should only be used in variable declarations; they should be referred to via variable everywhere else.
sass/components/_tabs.scss:21 [W] ColorVariable: Color literals like `transparent` should only be used in variable declarations; they should be referred to via variable everywhere else.

$ bundle exec scss-lint -v
scss-lint 0.34.0

$ bundle exec sass -v
Sass 3.4.13 (Selective Steve)
```

**Comment by sds** [comment]
Thanks for the repro case, @JohnAlbin.

It looks like this only happens in specific contexts where the Sass parser parses the tree as Sass script instead of a string, which occurs when a list literal contains function calls (`image-url` in your example).

Fixed in 83654ec9031ef7477cef012754a59aa47f299e53.
